### ISSUE TITLE
Access blocked: KDE Event Calendar Widget’s request is invalid

### ISSUE THREAD

**Issue by cocoepiblas** [Zren/plasma-applet-eventcalendar#331]
Operating System: KDE neon 5.26
KDE Plasma Version: 5.26.0
KDE Frameworks Version: 5.99.0
Qt Version: 5.15.6
Kernel Version: 5.15.0-52-generic (64-bit)
Graphics Platform: X11
Event Calendar 76

When I want to synchronize with Google Calendar, the Google page tells me:

You can’t sign in because KDE Event Calendar Widget sent an invalid request. You can try again later, or contact the [developer](https://accounts.google.com/) about this issue. [Learn more about this error](https://support.google.com/accounts/answer/12379384)
If you are a developer of KDE Event Calendar Widget, see [error details](https://accounts.google.com/).
Error 400: invalid_request

I used the command:

sudo systemctl restart sddm

It looked like it was going to work, but it took me to the previous message above.

**Comment by polrus** [comment]
I confirm the same error in Opnesuse and current version of widget downloaded from plasma widget browser

**Comment by CAG2Mark** [comment]
Looks like the authentication method being used before is now deprecated by Google and the following should be implemented instead: https://developers.google.com/identity/protocols/oauth2/native-app#redirect-uri_loopback

**Comment by lays147** [comment]
Same problem.

![image](https://user-images.githubusercontent.com/7799231/197356535-5c07ba32-9f06-4e78-9935-641832c67f30.png)

**Comment by adamth0mps0n** [comment]
Same issue here with Garuda Linux.

**Comment by mikhansd** [comment]
![image](https://user-images.githubusercontent.com/103534305/197418295-4ea20dc1-4f76-4cd9-af89-c50c2a376815.png)

"error details":
![image](https://user-images.githubusercontent.com/103534305/197418329-6c81e3eb-4451-4942-b5cc-4601f8df1334.png)

"Learn more about this error":
https://support.google.com/accounts/answer/12379384

It looks like Google started blocking OAuth flows in October 2022.

I had Event Calendar installed on a different machine before the requests were blocked, and Google Calendar still syncs on that calendar.

Both machines run Arch Linux.

**Comment by ksgremes** [comment]
To anyone that ends up here, I was able to circumvent this by using an old access token (from a configuration file in a previous installation I had in my disk), previously granted tokens should still work

The configuration is in: ~/.config/plasma-org.kde.plasma.desktop-appletsrc

In there you can search for "Google Calendar" and paste the token information. Then log out and back in
image
![image](https://user-images.githubusercontent.com/67567752/198590490-511c5f11-4516-417d-b230-4c02cea8c51e.png)

**Comment by cocoepiblas** [comment]
> To anyone that ends up here, I was able to circumvent this by using an old access token (from a configuration file in a previous installation I had in my disk), previously granted tokens should still work
> 
> The configuration is in: ~/.config/plasma-org.kde.plasma.desktop-appletsrc
> 
> In there you can search for "Google Calendar" and paste the token information. Then log out and back in image ![image](https://user-images.githubusercontent.com/67567752/198590490-511c5f11-4516-417d-b230-4c02cea8c51e.png)

Thanks for your help. In a previous installation I had in my disk with Manjaro, Event Calendar work with Google Calendar but this token does not appear. Event Calendar is not the last version.

**Comment by JackDinn** [comment]
Same has just happened to me. What is happening with this problem then? why is there no comment or fix after 2 weeks?

**Comment by CAG2Mark** [comment]
> Same has just happened to me. What is happening with this problem then? why is there no comment or fix after 2 weeks?

It's not trivial to fix. IIRC Google now requires clients like these to use a localhost server to retrieve the token.

Even if it were easy to fix, nobody has the obligation to fix it.

**Comment by JackDinn** [comment]
> > Same has just happened to me. What is happening with this problem then? why is there no comment or fix after 2 weeks?
> 
> It's not trivial to fix. IIRC Google now requires clients like these to use a localhost server to retrieve the token.
> 
> Even if it were easy to fix, nobody has the obligation to fix it.

Ok, i understand. So what should KDE users use for a simple tray calendar that syncs with google ATM? (not including anything that uses Akonadi which i dont like or use).

**Comment by cocoepiblas** [comment]
> > > Same has just happened to me. What is happening with this problem then? why is there no comment or fix after 2 weeks?
> > 
> > 
> > It's not trivial to fix. IIRC Google now requires clients like these to use a localhost server to retrieve the token.
> > Even if it were easy to fix, nobody has the obligation to fix it.
> 
> Ok, i understand. So what should KDE users use for a simple tray calendar that syncs with google ATM? (not including anything that uses Akonadi which i dont like or use).

What you ask could be useful, but in my case, what I would like to know should be something simple, and is where I locate the access token in my Manjaro, which has installed the EC v.72, to use it in my Neon EC v.76.

**Comment by Angel-arruda** [comment]
My newly Manjaro KDE Installation has only this snippet on that file :(   
[Containments][76][Applets][102][Configuration][Google Calendar]
calendarList=W10=
tasklistList=W10=

**Comment by mottasystem** [comment]
> > To anyone that ends up here, I was able to circumvent this by using an old access token (from a configuration file in a previous installation I had in my disk), previously granted tokens should still work
> > The configuration is in: ~/.config/plasma-org.kde.plasma.desktop-appletsrc
> > In there you can search for "Google Calendar" and paste the token information. Then log out and back in image ![image](https://user-images.githubusercontent.com/67567752/198590490-511c5f11-4516-417d-b230-4c02cea8c51e.png)
> 
> Thank you a lot! It's worked for me. Fresh installation have only few lines like calendar List=W10= tasklist List=W10= But old config file has tons of info that should be copied into new one.

Access tokens periodically expire and become invalid credentials for a related API request. - https://developers.google.com/identity/protocols/oauth2/native-app#redirect-uri_loopback

**Comment by Linux-Alex** [comment]
Please make a fix

**Comment by ak1ra26** [comment]
Same problem. Please, make a fix.

**Comment by etoven** [comment]
Confirmed same issue

**Comment by dwijonarko** [comment]
It's happened to me on my fresh install this plugin.

**Comment by sometimesabird** [comment]
Still no fix, Garuda Linux

**Comment by Linuxmaker** [comment]
Apart from the fact that the KDE applet is completely overloaded with information - I've already uninstalled it again - I'm looking for a reasonable way to easily enter Google calendar entries without a web browser on the Linux desktop (here Debian 11 with KDE Plasma). so that they also appear on smartphones. Is there a suitable software for Linux and KDE?

**Comment by Sadi58** [comment]
@Linuxmaker This one maybe?
[https://apps.kde.org/kalendar/](https://apps.kde.org/kalendar/)

**Comment by Linuxmaker** [comment]
> @Linuxmaker This one maybe? https://apps.kde.org/kalendar/

Thanks for the URL

**Comment by Quoteme** [comment]
Here is a fix:
1. `git clone https://github.com/gaganpreet/plasma-applet-eventcalendar.git eventcalendar`
2. `cd eventcalendar`
3. `git checkout google-calendar-fix`
4. `sh ./install`

use `sh ./uninstall` for uninstallation.

**Comment by etoven** [comment]
> This plasma applet has not been updated **since May 2022**, which means the developer is not able/willing to devote more time for this project. **There's no point in repeating here that this problem has not been fixed; instead, we apparently need a fork that will be developed more actively, for instance, by some of the existing 34 contributors.** I also suggest trimming it down to make it a bit easier to maintain, e.g. by excluding weather etc. for which there are already individual applets working very well, and keeping only _Google Calendar_ integration plus displaying date and time in a customizable format.

I use the weather info next to the calendar to book appts. That would have been unfortunate for me to do away with it.

**Comment by Sadi58** [comment]
> I use the weather info next to the calendar to book appts. That would have been unfortunate for me to do away with it.
@etoven Good point. If it can be maintained as it is, fine. My suggestion was just an idea to make it simpler to maintain by focusing on its main function. BTW I use a weather widget on both sides of the Event Calendar as shown below, but I don't have too many appointments as a retired person. ;-)
![Screenshot_2023-11-29_10-41-09](https://github.com/Zren/plasma-applet-eventcalendar/assets/6375934/f1771e52-b65b-464f-8908-bd9562fc8ddf)

**Comment by codemeleon** [comment]
> Here is a fix:
> 
>     1. `git clone https://github.com/gaganpreet/plasma-applet-eventcalendar.git eventcalendar`
> 
>     2. `cd eventcalendar`
> 
>     3. `git checkout google-calendar-fix`
> 
>     4. `sh ./install`
> 
> 
> use `sh ./uninstall` for uninstallation.

I think it is still broken.

**Comment by Sadi58** [comment]
@codemeleon Try this - after installing `golang` - I've been using it for a long time now: https://github.com/kanocz/plasma-applet-eventcalendar

And there's also this new update, which I haven't tried yet: https://github.com/ALikesToCode/plasma-applet-eventcalendar

**Comment by codemeleon** [comment]
@Sadi58, Thanks. I Noticed that allowing via Firefox fails. However, allowing through chromium based browser works.
